test/server/lib/DocApi.ts

```diff
@@ -160,33 +175,47 @@ describe('DocApi', function () {
         };
 
         home = await TestServer.startServer('home', tmpDir, suitename, additionalEnvConfiguration);
-        docs = await TestServer.startServer('docs', tmpDir, suitename, additionalEnvConfiguration, home.serverUrl);
         homeUrl = serverUrl = home.serverUrl;
+        docs = await TestServer.startServer('docs', tmpDir, suitename, additionalEnvConfiguration, homeUrl);
         hasHomeApi = true;
       });
       testDocApi({webhooksTestPort});
     });
 
     describe('behind a reverse-proxy', function () {
-      async function setupServersWithProxy(suitename: string, overrideEnvConf?: NodeJS.ProcessEnv) {
-        const proxy = new TestServerReverseProxy();
+      async function setupServersWithProxy(
+        suitename: string,
+        { withAppHomeInternalUrl }: { withAppHomeInternalUrl: boolean }
+      ) {
+        const proxy = await TestServerReverseProxy.build();
+
+        const homePort = await getAvailablePort(parseInt(process.env.GET_AVAILABLE_PORT_START || '8080', 10));
+        const home = new TestServer('home', homePort, tmpDir, suitename);
+
         const additionalEnvConfiguration = {
           ALLOWED_WEBHOOK_DOMAINS: `example.com,localhost:${webhooksTestPort}`,
           GRIST_DATA_DIR: dataDir,
-          APP_HOME_URL: await proxy.getServerUrl(),
+          APP_HOME_URL: proxy.serverUrl,
           GRIST_ORG_IN_PATH: 'true',
           GRIST_SINGLE_PORT: '0',
```

Comment:
I have taken a look. Actually, `devServerMain.ts` is not run in this case. If you take a look, it is only invoked when running the nbrowser tests.

Actually this variable is set for this piece of code:
https://github.com/gristlabs/grist-core/blob/1c26dd73fe7831115f58f3af844bf39ec921d982/app/server/lib/DocWorkerUtils.ts#L186

It is true that the use in `devServerMain` is confusing, but I don't want to change that part of the code because this PR is open for too long and I would like it to be merged.

Reply:
I resolve this conversation. Don't hesitate to ping me if you think this conversation is actually not over :)